File: src/types/nudge.ts
export type NudgeDismissal = {
  id: string;
  nudgeKey: string;
  scope?: string | null;
  dismissedForDate: string;
  createdAt: string;
};

export type ListNudgeDismissalsInput = {
  dismissedForDate: string;
  scope?: string | null;
};

export type DismissNudgeInput = {
  nudgeKey: string;
  scope?: string | null;
  dismissedForDate: string;
};

File: src-tauri/src/domain/nudge.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct NudgeDismissal {
    pub id: String,
    pub nudge_key: String,
    pub scope: Option<String>,
    pub dismissed_for_date: String,
    pub created_at: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ListNudgeDismissalsInput {
    pub dismissed_for_date: String,
    pub scope: Option<String>,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DismissNudgeInput {
    pub nudge_key: String,
    pub scope: Option<String>,
    pub dismissed_for_date: String,
}
